File: ClientApp/src/_models/Entity.ts
import { typeWithParameters } from "@angular/compiler/src/render3/util";

export class Entity {
    health: number;
    maxHealth: number;
    avatarID?: string;

    constructor(entityDTO: any) {
        this.health = entityDTO.health;
        this.maxHealth = entityDTO.maxHealth;
        this.avatarID = entityDTO.avatarID;
    }

    get HealthPercentage(): number {
        let percent = (this.health / this.maxHealth) * 100;
        if (percent < 0)
            percent = 0;
            
        return percent;
    }
}

File: Models/Character/Entity.cs
namespace taleOfDungir.Models
{
    public struct Entity
    {
        public long Health { get; set; }
        public long MaxHealth { get; set; }
        public long AvatarID { get; set; }

        public Entity(long health, long maxHealth, long avatarID)
        {
            Health = health;
            MaxHealth = maxHealth;
            AvatarID = avatarID;
        }

        public Entity(Character character){
            Health = character.Health;
            MaxHealth = character.MaxHealth;
            AvatarID = character.CharacterAvatarId;
        }
    }
}